### DISCUSSION TITLE
Two Camers

### DISCUSSION THREAD

**Discussion by swir00** [mainsail-crew/spyglass#86] [Q&A] [answered]
Is it possible to use two Picameras on RPI5

**Comment by mryel00** [comment]
With release 0.15 it's in theory possible. You need to create a second service, as one Spyglass instance can only run a stream for one cam at a time atm.

**Reply by swir00** [reply]
can you tell my how setup second service ?

**Reply by mryel00** [reply, marked as answer]
Doing a second service that also got a config, will be a bit more challenging, as you will need to edit the service files for that.
I'm expecting that you already got one service setup through `make install`. I'm also writing this without testing, so please report back, if you run into problems with this.

Run following commands:
```sh
sudo cp /etc/systemd/system/spyglass.service /etc/systemd/system/spyglass-2.service
cp -f ~/spyglass/resources/syglass.conf ~/printer_data/config/spyglass-2.conf
```
Now edit the `spyglass-2.service` with `sudo nano spyglass-2.service` and edite the `ExecStart=` to add `-c  /home/<your_pi_user>/printer_data/config/spyglass-2.conf` but change `<your_pi_user>` to the user of your pi. You can save with `ctrl+s` and exit with `ctrl+x`.
My user on the Pi is `mryel` so I would change the line to the following:
`ExecStart= /usr/local/bin/spyglass -c /home/mryel/printer_data/config/spyglass-2.conf`

After that you should have a second config in your UI. Change `CAMERA_NUM="0"` to `CAMERA_NUM="1"`, save it and run the following commands:
```sh
sudo systemctl daemon-reload
sudo echo "spyglass-2"  >> ~/printer_data/moonraker.asvc
sudo systemctl enable spyglass-2
sudo systemctl start spyglass-2
```

**Reply by swir00** [reply]
ok, I did it, I have one more question: what URL should I set? :)

**Reply by mryel00** [reply]
It doesn't matter. You just have to use a different port and adjust your cam settings in your UI accordingly.

**Reply by swir00** [reply]
something is wrong with the spyglass-2 service, it seems to be working but I can't get the preview, I did everything according to the instructions

**Reply by mryel00** [reply]
Did you try a reboot? If it doesn't help send the output of `systemctl status spyglass-2`

**Reply by swir00** [reply]
pi@klipper:~ $ systemctl status spyglass-2
● spyglass-2.service - spyglass - Picamera2 MJPG Streamer
     Loaded: loaded (/etc/systemd/system/spyglass-2.service; enabled; preset: enabled)
     Active: active (exited) since Tue 2024-08-27 15:12:56 CEST; 45s ago
       Docs: https://github.com/roamingthings/spyglass
    Process: 1004 ExecStart=c /home/pi/printer_data/config/spyglass-2.conf (code=exited, status=0/SUCCESS)
   Main PID: 1004 (code=exited, status=0/SUCCESS)
        CPU: 1ms

Aug 27 15:12:56 klipper systemd[1]: Started spyglass-2.service - spyglass - Picamera2 MJPG Streamer.
Aug 27 15:12:56 klipper (c)[1004]: spyglass-2.service: Executable c missing, skipping: No such file or directory

---------------
the file is in that directory, I checked it

**Reply by mryel00** [reply]
Send your service file. You most likely made a big mistake there

**Comment by swir00** [comment]
I did everything from scratch on a clean installation and it works, thanks for your help
